model/task/task.go

```diff
@@ -871,35 +884,240 @@ func (t *Task) ActivateTask(caller string) error {
 	t.ActivatedBy = caller
 	t.Activated = true
 	t.ActivatedTime = time.Now()
-	return UpdateOne(
+
+	return ActivateTasks([]Task{*t}, caller)
+}
+
+func ActivateTasks(tasks []Task, caller string) error {
+	taskIDs := make([]string, 0, len(tasks))
+	for _, t := range tasks {
+		taskIDs = append(taskIDs, t.Id)
+	}
+
+	_, err := UpdateAll(
 		bson.M{
-			IdKey: t.Id,
+			IdKey: bson.M{"$in": taskIDs},
 		},
 		bson.M{
 			"$set": bson.M{
 				ActivatedKey:     true,
 				ActivatedByKey:   caller,
-				ActivatedTimeKey: t.ActivatedTime,
+				ActivatedTimeKey: time.Now(),
 			},
 		})
+	if err != nil {
+		return errors.Wrap(err, "can't activate task")
+	}
+
+	for _, t := range tasks {
+		event.LogTaskActivated(t.Id, t.Execution, caller)
```

Comment:
We should choose a convention about whether or not we log regardless of error, since DeactivateDependencies does this differently.

Reply:
it looks to me like they both follow the same pattern: after the update was successful the event is logged.
Am I misunderstanding?